### PR TITLE
:sparkles:add kubekey provider

### PR DESCRIPTION
**Pull request by 24sama**
**What this PR does / why we need it**:
Add kubekey provider. The CAPKK will use ssh to bootstrap the cloud-init file in a running host.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: cmd/clusterctl/cmd/config_repositories_test.go [hunk 1]
```diff
@@ -207,6 +208,10 @@ var expectedOutputYaml = `- File: core_components.yaml
   Name: ibmcloud
   ProviderType: InfrastructureProvider
   URL: https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/releases/latest/
+- File: infrastructure-components.yaml
+  Name: outscale
```
Comment (oscr) [inline cmd/clusterctl/cmd/config_repositories_test.go:212]: Wrong name

**Comment by 24sama** [comment]
> For my understanding, is CAPKK a pure infrastructure provider or does it include node bootstrapping and control plane- functionality?

For now, there is a k3s bootstrap provider and k3s control plane provider in the project but we still have some unfinished things about those, like the provider name hhhh...

**Comment by killianmuldoon** [comment]
I'm also getting an error trying to use this:
```
 # The ssh IP address of the all instance. (e.g. "[{address: 192.168.100.3}, {address: 192.168.100.4}]")
export INSTANCES=<your-linux-ip-address>
```
export suggestion in the quickstart. clusterctl generates the yaml as expected, but it fails to unmarshal when using 'kubectl apply -f '. 

Once I fixed it up in the yaml manually it worked great though! I've got a capkk cluster 'running' (No underlying infra, just error messages :laughing:)

**Comment by 24sama** [comment]
Sorry to reply late @killianmuldoon.
Thanks for your feedback. I add the `KKZONE` to the quickstart, and I also modify the cluster-template.yaml that in the kubekey release page. Now they can generate the correct yaml file.

File: docs/book/src/user/quick-start.md [hunk 2]
```diff
@@ -593,7 +603,7 @@ before configuring a cluster with Cluster API. Instructions are provided for com
 Otherwise, you can look at the `clusterctl generate cluster` [command][clusterctl generate cluster] documentation for details about how to
 discover the list of variables required by a cluster templates.
 
-{{#tabs name:"tab-configuration-infrastructure" tabs:"AWS,Azure,CloudStack,DigitalOcean,Docker,Equinix Metal,GCP,IBM Cloud,Kubevirt,Metal3,Nutanix,OpenStack,Outscale,VCD,vcluster,Virtink,vSphere"}}
+{{#tabs name:"tab-configuration-infrastructure" tabs:"AWS,Azure,CloudStack,DigitalOcean,Docker,Equinix Metal,GCP,IBM Cloud,kubekey,Kubevirt,Metal3,Nutanix,OpenStack,Outscale,VCD,vcluster,Virtink,vSphere"}}
```
[the remark this answers is not in the document]
Comment (24sama) [reply docs/book/src/user/quick-start.md]: Yes, thanks to catch it.

**Comment by oscr** [comment]
@24sama Thank you! Great work. KubeKey looks like a really promising new provider. 

Pending accepted CLA
/lgtm

**Comment by 24sama** [comment]
Thanks for the review @oscr.

**Review by killianmuldoon — APPROVED** [review]
Looks good! Thanks for adding KubeKey  - looking forward to the ControlPlane and bootstrap providers.

/lgtm

**Comment by fabriziopandini** [comment]
lgtm pending CLA
/retest

**Comment by 24sama** [comment]
CLA has been passed.

**Comment by sbueringer** [comment]
Thx!

/approve

**Comment by 24sama** [comment]
Hi @sbueringer , can you call the bot to request an auto cherry pick?

**Comment by sbueringer** [comment]
@24sama Looks like it requires a manual cherry-pick. Feel free to open the same PR against release-1.2

**Pull request merged**